aws/resource_aws_batch_compute_environment.go

```diff
@@ -0,0 +1,428 @@
+				MinItems: 0,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"bid_percentage": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							ForceNew: true,
+						},
+						"desired_vcpus": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+						"ec2_key_pair": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+						"image_id": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+						"instance_role": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
+						"instance_type": {
+							Type:     schema.TypeSet,
+							Required: true,
+							ForceNew: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+							Set:      schema.HashString,
+						},
+						"max_vcpus": {
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+						"min_vcpus": {
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+						"security_group_ids": {
+							Type:     schema.TypeSet,
+							Required: true,
+							ForceNew: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+							Set:      schema.HashString,
+						},
+						"spot_iam_fleet_role": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+						"subnets": {
+							Type:     schema.TypeSet,
+							Required: true,
+							ForceNew: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+							Set:      schema.HashString,
+						},
+						"tags": tagsSchema(),
+						"type": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ForceNew:     true,
+							ValidateFunc: validation.StringInSlice([]string{"EC2", "SPOT"}, true),
+						},
+					},
+				},
+			},
+			"service_role": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"state": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{"ENABLED", "DISABLED"}, true),
+				Default:      "ENABLED",
+			},
+			"type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice([]string{"MANAGED", "UNMANAGED"}, true),
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"ecc_cluster_arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"status_reason": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsBatchComputeEnvironmentCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).batchconn
+
+	computeEnvironmentName := d.Get("compute_environment_name").(string)
+
+	log.Printf("[DEBUG] Create compute environment \"%s\".\n", computeEnvironmentName)
```

Comment:
To get a better idea about what is transiting, can you move this line 214, and reword it to:
`log.Printf("[DEBUG] Creating Batch Compute Environment: %s", input)`?

Since the `CreateComputeEnvironmentInput` struct has a `String` method, it will display it nicely, and provide much more information when debugging!